src/wallet/wallet_rpc_server.cpp

```diff
@@ -198,24 +199,77 @@ namespace tools
   bool wallet_rpc_server::run()
   {
     m_stop = false;
-    m_net_server.add_idle_handler([this](){
-      if (m_auto_refresh_period == 0) // disabled
+
+    const bool enable_auto_refresh = m_auto_refresh_period != 0;
+    const auto auto_refresh_evaluation_ms = std::chrono::milliseconds(200);
+
+    m_net_server.add_idle_handler([=] { // Implicit capture of this-pointer deprecated in C++20.
+      if (!enable_auto_refresh) // disabled
         return true;
-      if (boost::posix_time::microsec_clock::universal_time() < m_last_auto_refresh_time + boost::posix_time::seconds(m_auto_refresh_period))
+
+      // Check if m_auto_refresh_period seconds have passed since the last refresh attempt.
+      const auto auto_refresh_interval_ms = std::chrono::milliseconds(m_auto_refresh_period * 1'000);
+      if (auto_refresh_interval_ms <= auto_refresh_evaluation_ms)
+      {
+        LOG_PRINT_L0((boost::format(tr("The auto wallet sync evaluation interval of %i ms must be lager than the refresh interval of %i ms"))
+          % auto_refresh_evaluation_ms.count()
+          % auto_refresh_interval_ms.count()).str());
+        return true;
+      }
+
+      const auto now = std::chrono::system_clock::now();
```

Comment:
This kind of thing needs a monotonic clock (`steady_clock`), not the wall clock, as you want to measure elapsed time, not deltas between snapshots of the system clock, and they're not at all the same thing.

Reply:
Addressed